aws-api/src/main/java/com/amplifyframework/api/aws/AWSApiPlugin.java

```diff
@@ -102,6 +116,32 @@ public AWSApiPlugin(@NonNull ApiAuthProviders apiAuthProvider) {
         this.gqlApis = new HashSet<>();
         this.executorService = Executors.newCachedThreadPool();
         this.requestDecorator = new AuthRuleRequestDecorator(authProvider);
+        this.authProviderChains = null;
+        this.modelProvider = null;
```

Comment:
AWS API Plugin is already a beastly entity. We want to be thinking of ways to fan logic out of this facade class. Even today it has overgrown its Single Responsibility. Is there any way we can avoid putting these new actors right into the top level plugin class?

Reply:
Ahhh yes...I had that feeling as well, but since the purpose here was to prove out the concept itself, I went ahead with it. That being said, I have a couple ideas to clean this up moving forward.